src/field_control_avatar.c

```diff
@@ -68,7 +68,9 @@ static bool8 TryStartWarpEventScript(struct MapPosition *, u16);
 static bool8 TryStartMiscWalkingScripts(u16);
 static bool8 TryStartStepCountScript(u16);
 static void UpdateFriendshipStepCounter(void);
+#if OW_POISON_DAMAGE < GEN_5
 static bool8 UpdatePoisonStepCounter(void);
+#endif // OW_POISON_DAMAGE
```

Comment:
As a follow up to "we want to reduce the number of direct preproc branches" one could phrase this as
```c
if (OW_POISON_DAMAGE >= GEN_5) return;
```
in `UpdatePoisonStepCounter`.